File: Python/1451-1500/1480. Running Sum of 1d Array.py
# https://leetcode.com/problems/running-sum-of-1d-array/

class Solution:
    def runningSum(self, nums: List[int]) -> List[int]:
        prefix = nums[0]

        for index in range(1, len(nums)):
            prefix = nums[index] + prefix
            nums[index] = prefix

        return nums
        

File: Javascript/1451-1500/1480. Running Sum of 1d Array.js
// https://leetcode.com/problems/running-sum-of-1d-array/

var runningSum = function(nums) {
    let prefix = nums[0];

    for(let index=1; index<nums.length; index++){
        prefix = nums[index] + prefix;
        nums[index] = prefix;
    }

    return nums;
};
